### PR TITLE
Build separate full contributors list

### LINKED ISSUE #38749: List contributors in a single section not next to each PR in changelog
Thanks for trying this, @getdave. I think listing the contributors participating in the release will improve the release page. However, I am not sure listing them next to each PR is the way we should go. I think it adds too granular info to the changelog, making it harder to read and focus on the PRs themselves. Moreover, copying the list of PRs alone elsewhere (Google Docs, make.wordpress.org/core) will need an extra step (e.g., a regex) to remove each author.

My two cents is adding the list of contributors to the bottom would be a nice middle ground. Later on, other participants could be added, apart from the person merging each PR.

_Originally posted by @priethor in https://github.com/WordPress/gutenberg/pull/38704#discussion_r804787249_

### PR DESCRIPTION
**Pull request by getdave**
## Description

In https://github.com/WordPress/gutenberg/issues/38749 we learned of a few reasons why listing the usernames of users next to their respective PRs might not be a great idea.

This PR changes things up so we have a single list of contributors underneath a `Contributors` section. It looks like this for `12.5`

```
## Contributors

The following users merged PRs in this release:

@aaronrobertshaw @adamziel @ajlende @alanjacobmathew @alexstine @amustaque97 @andrewserong @annezazu @anton-vlasenko @arthur791004 @audrasjb @c4rl0sbr4v0 @carolinan @chad1008 @ciampo @cr0ybot @dcalhoun @dmsnell @fabiankaegy @fluiddot @fullofcaffeine @geriux @getdave @glendaviesnz @guarani @gziolo @hypest @imangm @jameskoster @jasmussen @jeremyfelt @jhnstn @johnbillion @jorgecontreras @jorgefilipecosta @kebbet @kevin940726 @KevinBatdorf @knilkantha @Mamaduka @mchowning @mkevins @ndiego @noisysocks @ntsekouras @oandregal @ocean90 @ockham @peterwilsoncc @ramonjd @ribaricplusplus @richtabor @ryelle @scruffian @SiobhyB @talldan @tellthemachines @vcanales @walbo @youknowriad
```

**Note**: I'm not sure if this will cause GH to auto-generate it's own list thus duplicating the list. @priethor this is one of the reasons why I avoided this approach in my previous PR.

Closes https://github.com/WordPress/gutenberg/issues/38749

## Testing Instructions

- Run `npm run changelog -- --milestone="Gutenberg 12.6` in Gutenberg repo root.
- See generated changelog.
- `First time contributors` should be "as was"
- Check `Contributors` section is there and includes a single entry per username.
- Check not "bot" users are listed in either `First time contributors` or `Contributors` sections.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by priethor — COMMENTED** [review]
Thanks for iterating on this, Dave!

Should we add an array of exclusions to avoid listing bots like `@dependabot[bot]`?

**Comment by getdave** [comment]
> Thanks for iterating on this, Dave!
> 
> Should we add an array of exclusions to avoid listing bots like `@dependabot[bot]`?

We should. If we're happy with this format then I'll refine this PR to do that cleanup.

**Comment by priethor** [comment]
The format works quite well, in my opinion. If after a few releases we see it doesn't, we can iterate iterate on it 👍

File: bin/plugin/commands/changelog.js
Comment (getdave) [inline bin/plugin/commands/changelog.js]: Note @priethor we are now skipping "bots" from the main changelog output itself. Do we want to do that?

File: bin/plugin/commands/changelog.js
Comment (getdave) [inline bin/plugin/commands/changelog.js:784]: Now handled in dedicated `skipUsers` function.

File: bin/plugin/commands/changelog.js
Comment (getdave) [inline bin/plugin/commands/changelog.js:822]: Sorting by lowercase required else those with capitals in usernames end up _before_ those which come alphabetically before them.

File: bin/plugin/commands/test/fixtures/pull-requests.json
Comment (getdave) [inline bin/plugin/commands/test/fixtures/pull-requests.json]: I've manually added this entry to the fixture data to simulate a PR by a "bot" user. This allows us to verify it is excluded in the final output.

**Review by bph — APPROVED** [review]
This is perfect for the contributors who delivered code!  @getdave :-) 
Not sure if that has any bearing on this PR. When I used the command in your test instructions, I got back the error message: "Cannot find milestone by title Gutenberg 12.5". I got the list when I switched to 12.6 so no problem. 

Which got me thinking, if that list is only available as long as there is not new release, we might need to consider storing it some where. If that's the Make Release block or the release notes on GitHub that's fine, we can copy/paste it into an overall list for the WordPress release. 

Awesome.

**Comment by getdave** [comment]
Thanks for your review @bph 🙇‍♂️ 

> Which got me thinking, if that list is only available as long as there is not new release

Once this PR is merged the next release (`12.7` maybe?) will show the list on the [`Releases` tab in Github](https://github.com/WordPress/gutenberg/releases) under the appropriate release.

This will then continue for all future releases meaning it will never be lost.

File: bin/plugin/commands/changelog.js
Comment (gziolo) [reply bin/plugin/commands/changelog.js]: Yes, that makes a lot of sense because we have now dependabot polluting the changelog 😅

File: bin/plugin/commands/changelog.js
Comment (gziolo) [inline bin/plugin/commands/changelog.js]: Nit: Maybe `skipCreatedByBots` for now? It was a bit unclear what it does in other places.

File: bin/plugin/commands/changelog.js
Comment (gziolo) [inline bin/plugin/commands/changelog.js:884]: Nice trick with trimming the output 👍

File: bin/plugin/commands/test/fixtures/pull-requests.json
Comment (gziolo) [reply bin/plugin/commands/test/fixtures/pull-requests.json]: Maybe we should keep it in a separate file and merge with the one autogenerated from an old release? The risk is that we would lose this custom entry if it gets regenerated in the future. It’s not a blocker but it’s something that we won’t have under control moving forward.

**Review by gziolo — APPROVED** [review]
Thank you for another iteration. Let’s give it a go and see how it looks on the release page. If there is too much duplication with the list of contributors added by GitHub we can always try to wrap the section in HTML comments and see if that helps.

I left some minor implementation notes to consider but not blockers.

File: bin/plugin/commands/changelog.js
Comment (priethor) [reply bin/plugin/commands/changelog.js]: The `skipUsers` function was inspired by #38538. Probably a rename would help clarify its purpose, such as `skipCreatedByBots` or `skipFakeContributors`.

File: bin/plugin/commands/changelog.js
Comment (priethor) [reply bin/plugin/commands/changelog.js]: Yes, we should do that 👍

File: bin/plugin/commands/changelog.js
Comment (getdave) [reply bin/plugin/commands/changelog.js]: Yeh actually for now renaming to include the "bots" term would be helpful. I was originally thinking it might be a catch all solution but there isn't a use case for that at the moment so let's keep things simple.

File: bin/plugin/commands/changelog.js
Comment (getdave) [reply bin/plugin/commands/changelog.js:884]: 🥳

File: bin/plugin/commands/changelog.js
Comment (getdave) [reply bin/plugin/commands/changelog.js]: ✅

File: bin/plugin/commands/test/fixtures/pull-requests.json
Comment (getdave) [reply bin/plugin/commands/test/fixtures/pull-requests.json]: ✅ Agreed. Done.

**Comment by getdave** [comment]
Will merge once all the tests pass.

**Pull request merged**
